service/controller/clusterapi/v27/resource/tcdp/create.go

```diff
@@ -31,7 +31,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "finding the tenant cluster's data plane cloud formation stack")
 
 		i := &cloudformation.DescribeStacksInput{
-			StackName: aws.String(legacykey.MainHostPreStackName(cr)),
+			StackName: aws.String(legacykey.StackNameCPI(cr)),
```

Comment:
Note that this here is the TCDP resource which is still in development and not used at all. There are more things to fix and test in the future. Just keeping it as it was.